go-controller/pkg/ovn/ovn.go

```diff
@@ -324,25 +324,49 @@ func (oc *Controller) ovnControllerEventChecker(stopChan chan struct{}) {
 	}
 }
 
+func podScheduledAndNetworked(pod *kapi.Pod) bool {
+	// Only care about scheduled and networked pods
+	return pod.Spec.NodeName != "" && !pod.Spec.HostNetwork
+}
+
+func podNamespacedName(pod *kapi.Pod) types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: pod.Namespace,
+		Name:      pod.Name,
+	}
+}
+
 // WatchPods starts the watching of Pod resource and calls back the appropriate handler logic
 func (oc *Controller) WatchPods() error {
+	handledPods := make(map[types.NamespacedName]bool)
```

Comment:
might be simpler to use `string(pod.UID)` as the key instead of a `types.NamespacedName`? (and then you can use `sets.String` too).

Reply:
good suggestion, done!